daemon/instructions.rs

```diff
@@ -131,14 +136,23 @@ fn enqueue(message: EnqueueMessage, state: &SharedState) -> Message {
             vec![TaskStatus::Stashed, TaskStatus::Locked],
         );
 
+
         for task_id in &matching {
-            state.change_status(*task_id, TaskStatus::Queued);
+            if message.enqueue_at.is_some() {
+                state.set_enqueue_at(*task_id, message.enqueue_at);
+            } else {
+                state.change_status(*task_id, TaskStatus::Queued);
+            }
         }
 
         (matching, mismatching)
     };
 
-    let message = "Tasks are enqueued";
+    let message = if message.enqueue_at.is_some() {
+        "Tasks enqueueing delayed"
+    } else {
+        "Tasks are enqueued"
```

Comment:
Ok. Here's what I'm thinking.

If you add an `enqueue_at` to a task and then `enqueue` or `start` it manually afterwards, before it has been enqueued automatically, the `enqueue_at` field will still be a `Some<DateTime>` instead of a `None`.
I guess this wouldn't be bad, since you only check `Stashed` tasks in `check_stashed`. However, it would make me feel better, if you could set `task.enqueue_at = None` right here.

While reviewing my own code I also found that the passthrough of a cloned task into `start_process` is super ugly. I guess I'll refactor this as soon as your PR is merged. I'll then add the `task.enqueue_at = None` to `start_process` myself :)

Reply:
I fixed the `enqueue` command by always setting it. It doesn't solve the same problem with the `start` command. However, I think you were going to address that in your refactor? Am I correct?

Reply:
Jep that's correct :)